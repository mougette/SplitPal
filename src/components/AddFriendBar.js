import React, { useState, useEffect, useContext } from 'react';
import {Post} from '../components/RestPost';
import { TouchableOpacity, TextInput, StyleSheet, Text, View } from "react-native";

const AddFriendBar = ( props ) => {
const [friend, setFriend] = useState("");
    return(
        <View style={styles.link}>
            <TextInput
                style={styles.input}
                onChangeText={setFriend}
                value={friend}
                placeholder={ props.search }
            />
            <TouchableOpacity onPress={() => Post("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_1/friend-request",
                                                             JSON.stringify({
                                                              userEmail: props.email,
                                                              friendEmail: friend,
                                                              }))}>
                <Text style={styles.blueTextButton}>{ props.button }</Text>
            </TouchableOpacity>
        </View>
    );

}

const styles = StyleSheet.create({
    link: {
        flexDirection: 'row',
        justifyContent: 'center',
        paddingTop: 20,
    },
    input: {
        width: '60%',
        borderWidth: 1,
        borderRadius: 7,
        padding: 10,
        backgroundColor: 'white',
    },
    blueTextButton: {
        fontSize: 20,
        borderWidth: 1,
        borderRadius: 7,
        overflow: "hidden",
        padding: 10,
        marginLeft: 10,
        color: '#2196F3',
        backgroundColor: 'white',
    },
});

export default AddFriendBar;