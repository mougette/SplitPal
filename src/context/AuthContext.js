import React, { useState} from "react";
import createDataContext from './createDataContext';
import {Post} from '../components/RestPost';
import {Get} from '../components/RestGet';

const authReducer = (state, action) => {
  switch (action.type) {
    case 'signout':
      return {token: null, email: ''};
    case 'signin':
    return {
            email: action.payload.email,
            token: action.payload.token,
          };
    case 'signup':
      return {
        token: action.payload.token,
        email: action.payload.email,
      };
  }
};

const signup = dispatch => {
    const [account, setAccount] = useState("");
  return ({email, password, password2, firstName, lastName, phoneNumber}) => {
    Post("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_1/login",
                JSON.stringify({
                 email: email,
                 password: password,
                 password2: password2,
                    firstName: firstName,
                    lastName: lastName,
                    phoneNumber: phoneNumber,
                 }))
        .then(response => {
    console.log("Account: " + account + " Account End");
      if(response == "Account Successfully Made") {
              dispatch({
                  type: 'signup',
                  payload: {
                      token: 'some access token here',
                      email: email,
                  },
              });
      }
      else {
              alert(response)
      }
      }
      );


  };
};

const signin = dispatch => {
  return ({email, password}) => {
    // Do some API Request here
   Get("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_1/login","?email="+email+"&password="+password)
   .then(response =>{
    if(response == "User info correct"){
    dispatch({
      type: 'signin',
      payload: {
        token: 'some access token here',
        email: email,
      },
    });
    }});
  };
};

const signout = dispatch => {
  return () => {
    dispatch({type: 'signout'});
  };
};

export const {Provider, Context} = createDataContext(
  authReducer,
  {signin, signout, signup},
  {token: null, email: ''},
);
