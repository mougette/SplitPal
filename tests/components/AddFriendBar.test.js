import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react-native'
import {Provider as AuthProvider} from '../../src/context/AuthContext.js';
import {Context as AuthContext} from '../../src/context/AuthContext';
import {navigate} from '@reach/router'
import AddFriendBar from '../../src/components/AddFriendBar';
import {NavigationContainer} from '@react-navigation/native';

global.fetch = jest.fn().mockImplementation(() => {
      var p = new Promise((resolve, reject) => {
        resolve({
          ok: true,
          Id: '123',
          json: function() {
            return {Id: '123'}
          }
        });
      });
      })
describe('Test Hooks', () => {

    it('navigates on button press', () => {
    const tree = renderer
        .create(<AddFriendBar/>);

    expect(tree).toMatchSnapshot();
    });

    it('click Button',  () => {
        const Post = jest.fn();
        const {getByText} = render(<AddFriendBar button="Add Friend" onPress ={Post}/>);
            fireEvent.press(getByText("Add Friend"));
            expect(Post).toBeCalledTimes(1);

    });

});