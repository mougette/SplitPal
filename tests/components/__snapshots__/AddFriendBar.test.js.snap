// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Hooks navigates on button press 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "center",
      "paddingTop": 20,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 7,
        "borderWidth": 1,
        "padding": 10,
        "width": "60%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 7,
          "borderWidth": 1,
          "color": "#2196F3",
          "fontSize": 20,
          "marginLeft": 10,
          "overflow": "hidden",
          "padding": 10,
        }
      }
    />
  </View>
</View>
`;
